sections/Deals/AcceptOrReject/AcceptOrRejectSection/AcceptOrRejectSection.tsx

```diff
@@ -434,102 +447,106 @@ const AcceptOrRejectDeal: FC<AcceptOrRejectDealProps> = ({
 			setTxState(TransactionStatus.FAILED);
 		}
 	}, [
-		deal?.poolAddress,
-		setTxState,
 		walletAddress,
 		signer,
-		monitorTransaction,
+		deal.poolAddress,
 		poolBalances?.purchaseTokenDecimals,
+		txType,
+		setTxState,
+		isMaxValue,
 		gasLimitEstimate,
 		gasPrice,
-		txType,
+		isEmptyInput,
 		inputValue,
-		isMaxValue,
+		monitorTransaction,
 		poolBalancesQuery,
 	]);
 
 	useEffect(() => {
 		async function getGasLimitEstimate() {
 			if (!walletAddress || !signer || !deal?.poolAddress || !poolBalances?.purchaseTokenDecimals)
 				return setGasLimitEstimate(null);
+
 			const contract = new ethers.Contract(deal.poolAddress, poolAbi, signer);
-			if (txType === TransactionType.Withdraw && isMaxValue) {
-				setGasLimitEstimate(wei(await contract.estimateGas.withdrawMaxFromPool(), 0));
-			} else if (txType === TransactionType.Withdraw) {
-				setGasLimitEstimate(
-					wei(
-						await contract.estimateGas.withdrawFromPool(
-							ethers.utils.parseUnits(
-								(inputValue ?? 0).toString(),
-								poolBalances?.purchaseTokenDecimals ?? 0
-							)
-						),
-						0
-					)
-				);
-			} else if (txType === TransactionType.Accept && isMaxValue) {
-				setGasLimitEstimate(wei(await contract.estimateGas.acceptMaxDealTokens(), 0));
-			} else if (txType === TransactionType.Accept) {
-				setGasLimitEstimate(
-					wei(
-						await contract.estimateGas.acceptDealTokens(
-							ethers.utils.parseUnits(
-								(inputValue ?? 0).toString(),
-								poolBalances?.purchaseTokenDecimals ?? 0
-							)
-						),
-						0
-					)
-				);
-			} else {
-				return setGasLimitEstimate(null);
-			}
+
+			const txTypeOptions = {
+				[TransactionDealType.Withdraw]: async () => {
+					if (isMaxValue) {
+						return setGasLimitEstimate(wei(await contract.estimateGas.withdrawMaxFromPool(), 0));
+					}
+
+					return setGasLimitEstimate(
+						wei(
+							await contract.estimateGas.withdrawFromPool(
+								ethers.utils.parseUnits(
+									(isEmptyInput ? 0 : inputValue).toString(),
+									poolBalances?.purchaseTokenDecimals ?? 0
+								)
+							),
+							0
+						)
+					);
+				},
+				[TransactionDealType.AcceptDeal]: async () => {
+					if (isMaxValue) {
+						return setGasLimitEstimate(wei(await contract.estimateGas.acceptMaxDealTokens(), 0));
+					}
+
+					return setGasLimitEstimate(
+						wei(
+							await contract.estimateGas.acceptDealTokens(
+								ethers.utils.parseUnits(
+									(isEmptyInput ? 0 : inputValue).toString(),
+									poolBalances?.purchaseTokenDecimals ?? 0
+								)
+							),
+							0
+						)
+					);
+				},
+			};
+
+			txTypeOptions[txType]();
```

Comment:
I think I wrote a comment about changing this? You replied "done". Not sure what happened.

Reply:
Didn't you say to remove the `typeof(txTypeOptions[txType]) === function`?

Reply:
I think it's a good way to make the code looks better than this: https://github.com/AelinXYZ/aelin-ui/pull/175/files/4febeebdd56f1ed8dfed97cd1db4341070d9c97b#diff-e2c2e3c0890698f3c98d9f95d199b5248863440a054b97a6ab9b26b54193d470L385-L420